core/src/test/scala/org/apache/spark/benchmark/Benchmark.scala

```diff
@@ -141,8 +141,9 @@ private[spark] class Benchmark(
     val minIters = if (overrideNumIters != 0) overrideNumIters else minNumIters
     val minDuration = if (overrideNumIters != 0) 0 else minTime.toNanos
     val runTimes = ArrayBuffer[Long]()
+    val startTime = System.nanoTime()
     var i = 0
-    while (i < minIters || runTimes.sum < minDuration) {
+    while (i < minIters || (System.nanoTime() - startTime) < minDuration) {
```

Comment:
Is this change related?

Reply:
I meant to call this out. My empty properties test case was taking way too long. I found that the overhead of the loop was orders of magnitude longer than the function itself, especially when the array buffer was needing to resize a lot.
So instead of summing the time for each run I changed it to keep track of the total time elapsed.
I doubt that it will affect many other benchmarks, but it is something to note.